crates/cranelift-shared/src/obj.rs

```diff
@@ -103,14 +104,17 @@ impl<'a> ModuleTextBuilder<'a> {
     pub fn append_func(
         &mut self,
         name: &str,
-        func: &'a CompiledFunction,
+        body: &[u8],
+        alignment: u32,
+        unwind_info: Option<&'a UnwindInfo>,
+        relocations: &[Relocation],
         resolve_reloc_target: impl Fn(FuncIndex) -> usize,
     ) -> (SymbolId, Range<u64>) {
```

Comment:
Not something that needs to be handled here in this PR, but in the future this will need to be invoked by winch and I _think_ that most of the arguments here can simply become the output of `MachBuffer`'s compilation step since I think it's largely all packaged in there already (except maybe alignment IIRC). Just something to keep in mind though as more of the winch bits are filled out.